### PR TITLE
servermaster(engine): cancel/delete job avoid return error when job is failed

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7315

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/jobmanager.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (jm *JobManagerImpl) CancelJob(ctx context.Context, req *pb.CancelJobReques
 	if err != nil {
 		return nil, err
 	}
-	if pbJob.State == pb.Job_Finished || pbJob.State == pb.Job_Canceled {
+	if pbJob.State == pb.Job_Finished || pbJob.State == pb.Job_Canceled || pbJob.State == pb.Job_Failed {
```
Comment (CharlesCheung96) [inline engine/servermaster/jobmanager.go:123]: Perhaps abstracting a method can make the meaning clearer.

File: engine/servermaster/errors.go [hunk 2]
```diff
@@ -35,11 +35,11 @@ type JobAlreadyExistsError struct {
 	JobID string
 }
 
-// ErrJobNotStopped indicates that a given job is not stopped.
-var ErrJobNotStopped = rpcerror.Normalize[JobNotStoppedError]()
+// ErrJobNotTerminated indicates that a given job is in terminated state.
```
Comment (amyangfei) [inline engine/servermaster/errors.go:38]:
```suggestion
// ErrJobNotTerminated indicates that a given job is not in terminated state.
```

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**